File: stringMingling/mingleString.py
def mingle(stringp,stringq):
    newString = ""
    if len(stringp) > len(stringq):
        n = len(stringp)
    else:
        n = len(stringq)

    for i in range(n):
        if i >= len(stringp):
            newString += stringq[i]
        elif i < len(stringp):
            newString += stringp[i] + stringq[i]

    
    for i in range(len(stringp)):
        if stringp[i] == "a":
            print("La letra " +stringp[i]+ " aparece por primera vez en la posicion: "+str(i))
            
    return newString
    
stringp = "abc"
stringq = "pqrst"
n = 0
print(mingle(stringp,stringq))

File: stringMingling/mingleString.java
package stringMingling;

public class mingleString {
    
    public static String mingleStr(String stringp, String stringq){
        String newString = "";
        int n = 0;

        if(stringp.length() > stringq.length()){
            n = stringp.length();
        }
        else{
            n = stringq.length();
        }

        for(int i=0;i<n;i++){
            if(i >= stringp.length()){
                newString += stringq.charAt(i);
            }
            else if(i < stringp.length()){
                newString += stringp.charAt(i);
                newString += stringq.charAt(i);
            }
        }
        return newString;
    }

    public static void main(String[] args) {
        String stringp = "abc";
        String stringq = "pqrst";
        System.out.println(mingleStr(stringp, stringq));
    }
}
